submissions/f3d0t/friends-app/app.js

```diff
@@ -0,0 +1,173 @@
+		this.currentCards = [...this.allCards];
+		this.renderCards();
+	}
+	filterByGender(gender) {
+		if (gender === "male") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard.gender === "male");
+		}
+		if (gender === "female") {
+			this.currentCards = this.allCards.filter((friendCard) => friendCard.gender === "female");
+		}
+		if (gender === "all") {
+			this.currentCards = [...this.allCards];
+		}
+		this.renderCards();
+	}
+	filterBySearch(searchValue) {
+		const filteredArray = this.currentCards.filter((friendCard) =>
+			friendCard.searchTarget.toLowerCase().includes(searchValue.toLowerCase())
+		);
+		this.renderCards(filteredArray);
+	}
+	sortByName(sortType) {
+		if (sortType === "az") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "name", "ascending");
+		}
+		if (sortType === "za") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "name", "descending");
+		}
+		this.renderCards();
+	}
+	sortByAge(sortType) {
+		if (sortType === "ascending") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "age", "ascending");
+		}
+		if (sortType === "descending") {
+			this.currentCards = sortObjectsByPropertyValue(this.currentCards, "age", "descending");
+		}
+		this.renderCards();
+	}
+}
+
+const sortObjectsByPropertyValue = (arrayOfObjects, key, sortOrder) => {
+	const array = arrayOfObjects;
+	array.sort((a, b) => {
+		if (a[key] > b[key]) {
+			return 1;
+		} else if (a[key] < b[key]) {
+			return -1;
+		} else {
+			return 0;
+		}
+	});
+	return sortOrder === "ascending" ? array : array.reverse();
+};
+
+class FriendCard {
+	constructor(friendData) {
+		this.gender = friendData.gender;
+		this.name = `${friendData.name.first} ${friendData.name.last}`;
+		this.age = friendData.dob.age;
+		this.photo = friendData.picture.large;
+		this.email = friendData.email;
+		this.phone = friendData.cell;
+		this.country = friendData.location.country;
+		this.searchTarget = `${this.name} ${this.age} ${this.email} ${this.phone} ${this.country}`;
+	}
+
+	createCard() {
+		const card = document.createElement("div");
+		card.classList.add("card");
+		card.innerHTML = `<img class="card__img" src="${this.photo}" alt="${this.name} photo"'>
+                          <h3 class="card__name">${this.name}</h3>
+                          <span class="card__age">Age: ${this.age}</span>
+                          <a class="card__phone" href="tel:+${this.phone.replace(/\D/g, "")}">${this.phone}</a>
+                          <a class="card__email" href="mailto:${this.email}">${this.email}</a>
+                          <span class="card__country">${this.country}</span>`;
+		return card;
+	}
+}
+
+const dayNightChange = () => {
+	BODY.classList.toggle("light");
+};
+
+document.addEventListener("DOMContentLoaded", () => {
+	fetchData(REQUEST_LINK).then((resultDataArray) => initApp(resultDataArray));
+	DAY_NIGHT.addEventListener("click", () => {
+		dayNightChange();
+	});
+});
```

Comment:
```suggestion
document.addEventListener("DOMContentLoaded", () => {
	fetchData(REQUEST_LINK).then(initApp);
	DAY_NIGHT.addEventListener("click", () => {
		BODY.classList.toggle("light");
	});
});
```

Reply:
wooow, It calls initApp already with fetched data as a parameter!! Cool